qiskit/providers/honeywell/honeywellprovider.py

```diff
@@ -0,0 +1,218 @@
+#   Copyright 2019-2020 Honeywell, Intl. (www.honeywell.com)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Provider for a Honeywell backends."""
+
+import logging
+from collections import OrderedDict
+
+from qiskit.providers import BaseProvider
+from qiskit.providers.models import BackendConfiguration
+
+from .api import HoneywellClient
+from .credentials import discover_credentials, write_creds_to_qiskitrc, remove_creds_from_qiskitrc
+from .exceptions import HoneywellCredentialsNotFound
+from .honeywellbackend import HoneywellBackend
+
+
+logger = logging.getLogger(__name__)
+
+
+class HoneywellProvider(BaseProvider):
+    """Provider for Honeywell backends."""
+
+    def __init__(self):
+        """Return a new HoneywellProvider."""
+        super().__init__()
+
+        # Get a connection to Honeywell.
+        self._api = HoneywellClient()
+
+        # Populate the list of remote backends.
+        self._backends = None
+
+    def load_account(self):
+        """ Obtain stored credentials """
+        credentials = discover_credentials()
+
+        if not credentials:
+            raise HoneywellCredentialsNotFound
+
+        self._api.authenticate(credentials)
+
+    def save_account(self, token, overwrite=False, filename=None):
+        """ Save the credentials onto disk """
+        self._api.authenticate(token)
+        if filename:
+            write_creds_to_qiskitrc(token, overwrite, filename)
+        else:
+            write_creds_to_qiskitrc(token, overwrite)
```

Comment:
Was the plan to reuse the qiskitrc that the ibmq provider is using? If so I'm not sure that will work because the ibmq qiskitrc tries to parse the entire config file: https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/credentials/configrc.py#L62-L74 (IMO this is a terrible way to do credentials, or use a config file, but at this point that interface is the defined behavior for the ibmq provider)

I guess it depends on what the default is defined as.

Reply:
That was the plan but that's a good point. I'll use a different config file (we also agree about the config file)